bin/weewx/gettext.py

```diff
@@ -0,0 +1,166 @@
+# Simple localization for WeeWX
+# Copyright (C) 2021 Johanna Karen Roedenbeck
+
+#    See the file LICENSE.txt for your full rights.
+
+"""
+  provides tag $gettext(key, page)
+  
+  Language dependent texts are stored in special localization files.
+  The are merged into skin_dict in reportengine.py according to the
+  language the user chooses in weewx.conf for the given report. 
+  Different reports can be in different languages.
+  
+  The file defines language dependent label texts as well as other
+  texts used in skins. 
+  
+  The values provided by $gettext(key, page) are found in the file
+  in the [Texts] section. Subsections in the [Texts] section can
+  provide special texts for certain templates. To get values from
+  subsections $gettext() is called with special tag $page for the
+  parameter page. That is especially useful in include files that
+  are included in multiple templates. 
+
+  Examples:
+  
+  Assume the localization file to be:
+  
+  [Texts]
+    Key1 = Text1
+    [[index]]
+      Title = Text2
+    [[othertemplate]]
+      Title = Text3
+  
+  With that file is:
+  
+  $gettext("Key1")           ==> Text1
+  
+  $gettext("Title",$page)    ==> is Text2 if included in template "index"
+                                 and Text3 if included in template 
+                                 "othertemplate"
+                                
+  $gettext(page=$page).Title ==> same as before
+  
+  You can use "speaking" key names. If they contain whitespace they need
+  to be enclosed in quotes in the localization file
+  
+  If the value contains commas you must enclose it in single or double
+  quotes. Otherwise an array instead of a string is returned.
+  
+"""
+
+from weewx.cheetahgenerator import SearchList
+
+# Test for new-style weewx v4 logging by trying to import weeutil.logger
+import weeutil.logger
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class Gettext(SearchList):
+
+    def __init__(self,generator):
+        """Create an instance of the class"""
+        super(Gettext,self).__init__(generator)
```

Comment:
If you don't need any logic in `Gettext.__init__()`, then it can be omitted entirely.

Reply:
Done.